governator-test-junit/README.md

```diff
@@ -0,0 +1,81 @@
+
+# Dependency
+Add the following to your build.gradle
+```
+testCompile 'com.netflix.governator:governator-test-junit:latest.release'
+```
+
+# Example
+```java
+@RunWith(GovernatorJunit4ClassRunner.class)                     //You must use the Runner for these features to work
+@ModulesForTesting({ MyModule.class, ArchaiusModule.class })    //Specify any Modules you wish to include in your test
+public class ExampleTest {
+    
+    @Inject
+    @ReplaceWithMock                                            //Indicate that you wish this dependency to be Mocked
+    SomeDependency dependency;  
+    
+    @Inject
+    @WrapWithSpy                                                //Indicate that you wish to wrap this dependency with a Spy
+    SomeRealDependency realDependency;                      
+    
+    @Before
+    public void setup() {
+        Mockito.when(dependency.getValue()).thenReturn("Test"); //Specify desired behavior for your Mock
+    }
+    
+    @Test
+    public void testMock() {
+        assertEquals("Test", dependency.getValue());            //Verify behavior of your Mock
+    }
+    
+    @Test
+    public void testSpy() {
+        realDependency.getValue();
+        Mockito.verify(realDependency, Mockito.times(1)).getValue(); //Verify that this object was interacted with exactly once
+    }
+    
+    @Inject
+    Config config;
+    
+    @Test
+    @TestPropertyOverride({"myProperty=test"})                    //Specify property values you wish to be set
+    public void testConfig() {
+        assertEquals("test", config.getString("myProperty"));    //Verify that your property was set as expected
+    }
+    
+    @Test
+    @TestPropertyOverride(propertyFiles={"testProps.properties"}) //Properties may also be loaded from a file
+    public void testConfigFromFile() {
+        assertEquals("test", config.getString("myProperty"));   //Verify that your property was set as expected
+    }
+}
+```
+
+# Controlling Injector Creation
+You may choose whether the Injector will be created once per test class or once per test method by setting the injectorCreation attribute of [@ModulesForTesting](https://github.com/Netflix/governator/blob/master/governator-test-junit/src/main/java/com/netflix/governator/guice/test/ModulesForTesting.java). **By default, the injector is created once per test class.** This should only be modified if your Injector is in some way stateful and that state has been modified by one of your tests.
```

Comment:
What's actually important is whether or not the injected object is mutable. If the injected object is mutable, set the `injectorCreation` attribute of `@ModulesForTesting` to have a new instance of the `Injector` per test method.